### PR TITLE
Bugfix/jira 548

### PR DESCRIPTION
**Pull request by TayGov**
Adding missing code from webMethods. Updated order and added tests to cover.
image

### CODE REVIEW

**Review by adamkroonbcgov — DISMISSED** [review]

File: src/main/java/ca/bc/gov/iamp/bcparis/processor/datagram/DriverProcessor.java [hunk 1]
```diff
@@ -40,20 +40,20 @@ public Layer7Message process(Layer7Message message) {
...
 			log.info("Driver message processing completed.");
 			return message;
 				
 		}catch (ICBCRestException e) {
 			String content = messageService.parseResponseError(e.getResponseContent());
-			content = parseDriverResponse(content);
-			content = messageService.buildResponse(body, content);
+			content =  messageService.parseDriverResponse(content);
+			content = messageService.buildDriverResponse(body, content);
 			body.setMsgFFmt(content);
```
Comment (alexjoybc) [inline src/main/java/ca/bc/gov/iamp/bcparis/processor/datagram/DriverProcessor.java:57]: I don't understand why we are setting `MsgFFmt` before throwing rethrowing the `ICBCRestException`

File: src/main/java/ca/bc/gov/iamp/bcparis/processor/datagram/PORProcessor.java [hunk 2]
```diff
@@ -58,7 +58,7 @@ public Layer7Message process(Layer7Message message) {
 		}catch (PORRestException e) {
 			String content = messageService.parseResponseError(e.getResponseContent());
 			content = messageService.escape(content);
-			content = messageService.buildResponse(body, content);
+			content = buildResponse(message, content);
 			body.setMsgFFmt(content);
```
Comment (alexjoybc) [inline src/main/java/ca/bc/gov/iamp/bcparis/processor/datagram/PORProcessor.java:62]: Same here, why setting the MsgFFmt before throwing the exception

File: src/main/java/ca/bc/gov/iamp/bcparis/processor/datagram/VehicleProcessor.java [hunk 3]
```diff
@@ -55,8 +55,8 @@ public Layer7Message process(Layer7Message message) {
 			
 		}catch (ICBCRestException e) {
 			String content = messageService.parseResponseError(e.getResponseContent());
-			content = parseVehicleResponse(content);
-			content = messageService.buildResponse(body, content);
+			content = messageService.parseVehicleResponse(content);
+			content = messageService.buildVehicleResponse(body, content);
 			body.setMsgFFmt(content);
 			throw e;
```
Comment (alexjoybc) [inline src/main/java/ca/bc/gov/iamp/bcparis/processor/datagram/VehicleProcessor.java:61]: Setting msg before throwing exception

File: src/main/java/ca/bc/gov/iamp/bcparis/service/MessageService.java [hunk 4]
```diff
@@ -34,25 +34,14 @@
...
 
-	private String buildResponse(String from, String to, String text, String re,Body body, String icbcResponse) {
+	public String buildResponse(final Body body, final String icbcResponse) {
+		final String from = this.parseResponse(body.getCDATAAttribute("FROM"));
+		final String to = this.parseResponse(body.getCDATAAttribute("TO"));
+		final String text = this.parseResponse(body.getCDATAAttribute("TEXT"));
+		final String re = this.parseResponse(body.getCDATAAttribute("RE"));
 		return schema
-				.replace("${from}", to)
+				.replace("${from}", to) // Why though
```
Comment (adamkroonbcgov) [inline src/main/java/ca/bc/gov/iamp/bcparis/service/MessageService.java]: Maybe throw a TODO on this? :P

File: src/main/java/ca/bc/gov/iamp/bcparis/service/MessageService.java [hunk 5]
```diff
@@ -34,17 +34,15 @@
...
-		
+		final String receiver = this.parseResponse(body.getCDATAAttribute("FROM")); //This becomes the receiver of the message
+		final String sender = this.parseResponse(body.getCDATAAttribute("TO")); //This will become the sender
+		final String text = this.parseResponse(body.getCDATAAttribute("TEXT"));
+		final String re = this.parseResponse(body.getCDATAAttribute("RE"));
 		return schema
-				.replace("${from}", to)
-				.replace("${to}", from)
+				.replace("${from}", sender)
+				.replace("${to}", receiver)
```
Comment (alexjoybc) [inline src/main/java/ca/bc/gov/iamp/bcparis/service/MessageService.java:45]: thanks

**Review by alexjoybc — APPROVED** [review]
Thanks @TayGov , could you log couple of issues with the code?

**Pull request merged**